esp-wifi/src/esp_now/mod.rs

```diff
@@ -276,16 +275,32 @@ impl Debug for ReceivedData {
     }
 }
 
-/// A token used to create an `EspNow` instance while Wi-Fi is enabled.
-pub struct EspNowWithWifiCreateToken {
-    _private: (),
-}
+/// The interface to use for this peer
+#[derive(Debug, Clone, Copy, PartialEq)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub enum EspNowWifiInterface {
+    /// Use the AP interface
+    Ap,
+    /// Use the STA interface
+    Sta,
+}
+
+impl EspNowWifiInterface {
+    fn as_wifi_interface(&self) -> wifi_interface_t {
+        match self {
+            EspNowWifiInterface::Ap => wifi_interface_t_WIFI_IF_AP,
+            EspNowWifiInterface::Sta => wifi_interface_t_WIFI_IF_STA,
+        }
+    }
 
-/// Enables ESP-NOW while keeping Wi-Fi active.
-pub fn enable_esp_now_with_wifi(
-    device: crate::hal::peripherals::WIFI,
-) -> (crate::hal::peripherals::WIFI, EspNowWithWifiCreateToken) {
-    (device, EspNowWithWifiCreateToken { _private: () })
+    fn from_wifi_interface(interface: wifi_interface_t) -> Self {
+        #[allow(non_upper_case_globals)]
+        match interface {
+            wifi_interface_t_WIFI_IF_AP => EspNowWifiInterface::Ap,
+            wifi_interface_t_WIFI_IF_STA => EspNowWifiInterface::Sta,
+            _ => EspNowWifiInterface::Sta,
+        }
+    }
```

Comment:
Probably nit: But this can be simplified a bit
```suggestion
   fn from_wifi_interface(interface: wifi_interface_t) -> Self {
        #[allow(non_upper_case_globals)]
        match interface {
            wifi_interface_t_WIFI_IF_AP => EspNowWifiInterface::Ap,
            _ => EspNowWifiInterface::Sta,
        }
    }
```
Out of curiosity, is there any reason why you've chosen to go with `EspNowWifiInterface::Ap` compared with `Self::Ap`?

Reply:
> Out of curiosity, is there any reason why you've chosen to go with EspNowWifiInterface::Ap compared with Self::Ap?

Pretty much a question of preference/taste. I don't think there is real consensus in the community ( https://users.rust-lang.org/t/idiomatic-use-of-self/73754/1 ) and it's what RA generates by default (and what I personally see the most in other's code - and apparently prefer myself because to me it looks more readable and to the point)

But maybe we want to mandate either the full-enum name or Self in such situations in a code-style document? But if we do, we should somehow check it